emily/lambda/src/operations/deposits.rs

```diff
@@ -2,184 +2,449 @@
-    common::deserialize_request::<models::CreateDepositRequestContent>(body).map(|request| models::CreateDepositResponseContent {
-                bitcoin_txid: request.bitcoin_txid.to_string(),
-                bitcoin_tx_output_index: request.bitcoin_tx_output_index,
-                recipient: "MOCK_RECIPIENT".to_string(),
-                amount: 11111.0,
-                status: emily::models::OpStatus::Pending,
-                status_message: "MOCK_CREATE_DEPOSIT_RESPONSE".to_string(),
-                parameters: Box::new(models::DepositParameters {
-                    lock_time:22222.0,
-                    max_fee: 33333.0,
-                    reclaim_script: "MOCK_RECLAIM_SCRIPT".to_string()
-                }),
-                last_update_block_hash: None, // Unknown at creation
-                last_update_height: None, // Unknown at creation
-                fulfillment: None, // Unknown at creation
-            })
-        .and_then(|response| {
-            common::package_response(response, 201)
-        })
+
+    // Extract request parameters.
+    let request: models::CreateDepositRequestContent =
+        common::deserialize_request::<models::CreateDepositRequestContent>(body)?;
+
+    // TODO:
+    // Change "reclaim script" phrasing to be "deposit script" in consistent locations.
+    let reclaim_script = request.reclaim.clone();
+    let deposit_script_fields: FieldsExtractedFromDepositScript = extract_fields_from_reclaim_script(reclaim_script.clone())?;
+
+    // TODO:
+    // Get current latest stacks block from chainstate table.
+    let stacks_block_hash: String = "DUMMY_BLOCK_HASH".to_string();
+    let stacks_block_height: u64 = 44444;
+
+    // Create table entry.
+    let op_status: models::OpStatus = models::OpStatus::Pending;
+    let message: String = "Waiting for pickup by sBTC Bootstrap Signers.".to_string();
+
+    let deposit_table_entry: DepositTableEntry = DepositTableEntry {
+        bitcoin_txid: request.bitcoin_txid.clone(),
+        bitcoin_tx_output_index: request.bitcoin_tx_output_index as u16,
+        recipient: deposit_script_fields.recipient,
+        amount: deposit_script_fields.amount,
+        last_update_block_hash: stacks_block_hash.clone(),
+        last_update_height: stacks_block_height,
+        op_status,
+        history: vec![
+            DepositHistoryEntry {
+                op_status,
+                message,
+                stacks_block_height,
+                stacks_block_hash,
+            }
+        ],
+        version: 0,
+        max_fee: deposit_script_fields.max_fee,
+        lock_time: deposit_script_fields.lock_time,
+        reclaim_script,
+        fulfillment_bitcoin_txid: None,
+        fulfillment_bitcoin_tx_index: None,
+        fulfillment_btc_fee: None,
+        stacks_txid: None,
+    };
+
+    // Put entry into table.
+    context.dynamodb_client
+        .put_item()
+        .table_name(context.settings.deposit_table_name.clone())
```

Comment:
I think you can just use a reference here instead of cloning it. It looks more appealing that way to me.